### PR TITLE
[wip] forum channels

### LINKED ISSUE #593: implement `ForumChannel`
(note:
1. i wanna work on this
2. reference pr: discord/discord-api-docs#4761
)

### PR DESCRIPTION
**Pull request by MaskDuck**
## Summary

 forum channels discord/discord-api-docs#4761, closes #593 

### plans
~~- [ ] 1. make a new abc called `Threadable`~~
~~- [ ] 2. `TextChannel` and`ForumChannel` inherit them~~

updated plan:
- [ ] 1. make a new `ThreadableChannel`
- [ ] 2. make TextChannel and ForumChannel inherit them
## Checklist

- [ ] If code changes were made then they have been tested.
    - [ ] I have updated the documentation to reflect the changes.
- [ ] This PR fixes an issue.
- [x] This PR adds something new (e.g. new method or parameters).
- [x] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [ ] This PR is **not** a code change (e.g. documentation, README, ...)

### CODE REVIEW

**Comment by ooliver1** [comment]
could you use the `fixes ...` syntax for your relevant issue?

File: nextcord/types/channel.py [hunk 1]
```diff
@@ -74,6 +74,10 @@ class NewsChannel(_BaseGuildChannel, _TextChannelOptional):
     type: Literal[5]
 
 
+class ForumChannel(_BaseGuildChannel, _TextChannelOptional):
```
Comment (ooliver1) [inline nextcord/types/channel.py:77]: do forum channels have these from `_TextChannelOptional`?
```
    last_message_id: Optional[Snowflake]
    last_pin_timestamp: str
    rate_limit_per_user: int
```

File: nextcord/channel.py [hunk 2]
```diff
@@ -64,6 +64,7 @@
     'CategoryChannel',
     'GroupChannel',
     'PartialMessageable',
+    'ForumChannel'
```
Comment (ooliver1) [inline nextcord/channel.py:67]: trailing commas do help with git diffs

**Comment by MaskDuck** [comment]
updated plans 👍

**Comment by MaskDuck** [comment]
note: i deleted the old branch because the old implementation was fucked up

**Pull request closed without merging**